### ISSUE TITLE
Gingko reports suite failed, but all specs pass?

### ISSUE THREAD

**Issue by lefurgy** [onsi/ginkgo#520]
What would cause "Test Suite Failed" to be printed when all Specs are reported "Passed"?
No explanation of why the suite failed is reported.
It happens about 1 out of 10 times. The other 9 times report "Test Suite Passed".

go version: go1.9.4 linux/amd64
ginkgo: commit 11459a886d9cd66b319dac7ef1e917ee221372c9
             date:   Mon Aug 28 19:22:21 2017 -0600

Here is an example of the good output, followed by the bad output. (Sorry for the log messages that some of the specs print...I know it makes the output more confusing.)

Good output:
```
% ginkgo   -randomizeAllSpecs -progress --race -skipPackage=shared-diagnostics,shared-eventing,shared-logger,apis -cover -covermode atomic -nodes 16
Running Suite: Service Suite
============================
Random Seed: 1536803623 - Will randomize all specs
Will run 79 specs

Running in parallel across 16 nodes

\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:48.988Z","CALLER":"<autogenerated>:1","MESSAGE":"The total power budget or the power limits of some circuit breakers are below the Pcapmin of the servers that they govern. node=top budget=500","AUDIT":true}

\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:48.989Z","CALLER":"<autogenerated>:1","MESSAGE":"should not add children to unmanaged device type node a","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:48.996Z","CALLER":"<autogenerated>:1","MESSAGE":"could not initialize power policy (DCMI capping off) BMC=: unknown command","AUDIT":true}

\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:48.999Z","CALLER":"<autogenerated>:1","MESSAGE":"SENSOR ERROR: unreasonable average. sensor: {Metadata:0xc4203be3c0 value:4000000 Min:4000000 Max:4000000 Acc:4000000 Updates:1 timestamp:{wall:0 ext:63085005000 loc:<nil>} DependentSensors:[] MetricID:power DeviceName:system historyValue:[4000000] historyTime:[{wall:0 ext:63085005000 loc:<nil>}] historyIndex:0 historyLastIndex:0 count:1}  metadata: {Name:power Unit:W Scale:1 Kind:0 History:1 Tags:[5 10]}","AUDIT":true}

\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:48.999Z","CALLER":"<autogenerated>:1","MESSAGE":"senseserver BMC= unknown model fakemodel","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:49.001Z","CALLER":"<autogenerated>:1","MESSAGE":"cannot add children to server device a - 0","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:49.002Z","CALLER":"<autogenerated>:1","MESSAGE":"receptacle 1 is configured twice for enclosure fake. Check power topology configuration.","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:49.005Z","CALLER":"<autogenerated>:1","MESSAGE":"0 is not a valid PSU receptacle (1 to 10) for enclosure fake. Check power topology configuration.","AUDIT":true}

\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:49.004Z","CALLER":"<autogenerated>:1","MESSAGE":"cannot add children to server device a","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022
Ran 79 of 79 Specs in 2.034 seconds
SUCCESS! -- 79 Passed | 0 Failed | 0 Pending | 0 Skipped 

Ginkgo ran 1 suite in 8.950421982s
Test Suite Passed
```
Bad output:
```
% ginkgo   -randomizeAllSpecs -progress --race -skipPackage=shared-diagnostics,shared-eventing,shared-logger,apis -cover -covermode atomic -nodes 16
Running Suite: Service Suite
============================
Random Seed: 1536803633 - Will randomize all specs
Will run 79 specs

Running in parallel across 16 nodes

\u2022\u2022\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:59.358Z","CALLER":"<autogenerated>:1","MESSAGE":"0 is not a valid PSU receptacle (1 to 10) for enclosure fake. Check power topology configuration.","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:59.365Z","CALLER":"<autogenerated>:1","MESSAGE":"senseserver BMC= unknown model fakemodel","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:59.374Z","CALLER":"<autogenerated>:1","MESSAGE":"could not initialize power policy (DCMI capping off) BMC=: unknown command","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:59.381Z","CALLER":"<autogenerated>:1","MESSAGE":"receptacle 1 is configured twice for enclosure fake. Check power topology configuration.","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:59.388Z","CALLER":"<autogenerated>:1","MESSAGE":"should not add children to unmanaged device type node a","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:59.391Z","CALLER":"<autogenerated>:1","MESSAGE":"SENSOR ERROR: unreasonable average. sensor: {Metadata:0xc4203d02d0 value:4000000 Min:4000000 Max:4000000 Acc:4000000 Updates:1 timestamp:{wall:0 ext:63085005000 loc:<nil>} DependentSensors:[] MetricID:power DeviceName:system historyValue:[4000000] historyTime:[{wall:0 ext:63085005000 loc:<nil>}] historyIndex:0 historyLastIndex:0 count:1}  metadata: {Name:power Unit:W Scale:1 Kind:0 History:1 Tags:[5 10]}","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:59.394Z","CALLER":"<autogenerated>:1","MESSAGE":"The total power budget or the power limits of some circuit breakers are below the Pcapmin of the servers that they govern. node=top budget=500","AUDIT":true}

\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:59.392Z","CALLER":"<autogenerated>:1","MESSAGE":"cannot add children to server device a","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022\u2022
------------------------------
{"PRIORITY":"err","MSG_timestamp":"2018-09-13T01:53:59.398Z","CALLER":"<autogenerated>:1","MESSAGE":"cannot add children to server device a - 0","AUDIT":true}

\u2022\u2022\u2022\u2022\u2022\u2022\u2022
Ran 79 of 79 Specs in 2.026 seconds
SUCCESS! -- 79 Passed | 0 Failed | 0 Pending | 0 Skipped 

Ginkgo ran 1 suite in 8.756410679s
Test Suite Failed
```

**Comment by lefurgy** [comment]
Additional note: I have noticed that sometimes a race condition is reported in my code.

**Comment by onsi** [comment]
I'm working through the backlog of old Ginkgo issues - apologies as this issue is probably stale now.

I don't think I've ever seen this before - if you're still seeing it please let me know.

I'm going to close this for now, but feel free to reopen it if you'd like.

**Comment by fracasula** [comment]
This happens to us ([RudderStack](https://github.com/rudderlabs/rudder-server)) all the time when running tests with `-p`. The failure someone gets swallowed and ginkgo doesn't print the output. Removing `-p` makes the output magically appear.

This is how we run the tests: https://github.com/rudderlabs/rudder-server/blob/16cc6433d20e0966d5f3c303ee3f58ad1649af92/Makefile#L14

**Comment by anz-rfc-v2** [comment]
We experienced a similar symptom where ginkgo reports the suite had failed, where all specs passed.

It turns out we were calling `RunSpecs` twice within the same suite. Removing the second `RunSpecs` call resolved this issue.

**Comment by onsi** [comment]
hey all - i'm closing this out again (sorry, i've been behind on issues).  if folks are still seeing this or something related please open a new issue.
